### PR TITLE
Adding build arg for the alpine version the docker image is based on

### PR DESCRIPTION
**Pull request by arschles**
## What is the problem I am trying to address?

The final Athens docker image is based on `alpine`, which is the latest. We've had requests to pin it to a specific version

## How is the fix applied?

I've added a Docker build arg and pinned it to 

## What GitHub issue(s) does this PR fix or close?

Fixes https://github.com/gomods/athens/issues/1584

cc/ @dylan-bourque

### CODE REVIEW

File: cmd/proxy/Dockerfile [hunk 1]
```diff
@@ -16,7 +16,8 @@ ARG VERSION="unset"
 
 RUN DATE="$(date -u +%Y-%m-%d-%H:%M:%S-%Z)" && GO111MODULE=on CGO_ENABLED=0 GOPROXY="https://proxy.golang.org" go build -ldflags "-X github.com/gomods/athens/pkg/build.version=$VERSION -X github.com/gomods/athens/pkg/build.buildDate=$DATE" -o /bin/athens-proxy ./cmd/proxy
 
-FROM alpine
+ARG ALPINE_VERSION=3.9.5
+FROM alpine:3.9.5
```
Comment (dylan-bourque) [inline cmd/proxy/Dockerfile]: Shouldn't this line be `FROM alpine:${ALPINE_VERSION}` to make use of the build arg?

File: cmd/proxy/Dockerfile [hunk 2]
```diff
@@ -16,7 +16,8 @@ ARG VERSION="unset"
 
 RUN DATE="$(date -u +%Y-%m-%d-%H:%M:%S-%Z)" && GO111MODULE=on CGO_ENABLED=0 GOPROXY="https://proxy.golang.org" go build -ldflags "-X github.com/gomods/athens/pkg/build.version=$VERSION -X github.com/gomods/athens/pkg/build.buildDate=$DATE" -o /bin/athens-proxy ./cmd/proxy
 
-FROM alpine
+ARG ALPINE_VERSION=3.9.5
```
Comment (dsexton) [inline cmd/proxy/Dockerfile]: This `ARG` here is unused, however I would just drop it and leave the image tag explicit as you do on the next line `FROM alpine:3.9.5` or `FROM alpine:3.9`. 

Unless you are building a factory for building docker images I would just be explicit :)

File: cmd/proxy/Dockerfile (hunk 1 above)
Comment (arschles) [reply cmd/proxy/Dockerfile]: Thanks for that catch @dylan-bourque

File: cmd/proxy/Dockerfile (hunk 2 above)
Comment (twexler) [reply cmd/proxy/Dockerfile]: Agreed, this should just be removed.

Comment (arschles) [reply cmd/proxy/Dockerfile]: @dsexton I'm gonna leave in the arg (after fixing https://github.com/gomods/athens/pull/1585#discussion_r395838521) on the off chance someone wants to do a custom build of the image. Some folks have wanted to do the same for the [go version](https://github.com/gomods/athens/blob/e89cf0feb2155f60ab2b0a808a75c1c84ccf8e34/cmd/proxy/Dockerfile#L8)

Comment (marwan-at-work) [inline cmd/proxy/Dockerfile]: might as well upgrade it to 3.11?

Comment (arschles) [reply cmd/proxy/Dockerfile]: @marwan-at-work done, I upgraded to 3.11.5

**Pull request merged**